### DISCUSSION TITLE
Read image from memory buffer

### DISCUSSION THREAD

**Discussion by AlOa** [Tencent/ncnn#4547] [General]
Hi,
now simpleocv allow to read images only from file, but stb library allows to read image from memory buffer, so adding a function that allow to read images form memory is relative easy. I added this function fora  personal project, I simple added a function with signature:

`NCNN_EXPORT Mat imread(unsigned char *buf, int len, int flags = IMREAD_COLOR);`

to simpleocv.h and I copied the implementation of imread replacing stbi_load with stbi_load_from_memory. Do you thing that is something interenting to have?
                    Alberto

**Comment by nihui** [comment]
```cpp
NCNN_EXPORT Mat imdecode(const std::vector<uchar>& buf, int flags = IMREAD_COLOR);
```

There is imdecode in opencv. Follow the name.
If you have implemented it, contributions are welcome

**Comment by AlOa** [comment]
Hi, 
I made a pull request with the new function. It give a bunch of error, but some of then do not seems related with my code.

              Alberto
